zilliqa/src/consensus.rs

```diff
@@ -1294,39 +1552,20 @@ impl Consensus {
                 bytes
             })
             .collect();
-        let messages: Vec<_> = messages.iter().map(|m| m.as_slice()).collect();
+        let _messages: Vec<_> = messages.iter().map(|m| m.as_slice()).collect();
 
-        let committee = self.committee()?;
-        let public_keys: Vec<_> = agg
+        let _public_keys: Vec<_> = agg
             .signers
             .iter()
             .map(|i| committee.get_by_index(*i as usize).unwrap().public_key)
             .collect();
 
-        verify_messages(agg.signature, &messages, &public_keys)
-    }
-
-    fn get_leader(&self, view: u64) -> Result<Validator> {
-        // currently it's a simple round robin but later
-        // we will select the leader based on the weights
-        // Get the previous block, so we know the committee, then calculate the leader from there.
-        let block = self.get_block_by_view(view - 1)?;
-
-        let block = match block {
-            Some(block) => block,
-            None => self
-                .get_block_by_view(0)?
-                .ok_or_else(|| anyhow!("missing genesis block!"))?,
-        };
-
-        Ok(block.committee.leader(view))
+        // TODO: Implement batch verification - this will not work atm.
+        //verify_messages(agg.signature, &messages, &public_keys)
```

Comment:
aggregate QC would fail here, so it is commented out so that the unhappy path can be tested/used